src/commonMain/kotlin/au/id/micolous/metrodroid/card/iso7816/ISO7816Application.kt

```diff
@@ -54,8 +54,14 @@ class ISO7816ApplicationMutableCapsule(val appFci: ImmutableByteArray?,
     suspend fun dumpFileSFI(protocol: ISO7816Protocol, sfi: Int, recordLen: Int): ISO7816File? {
         val data = try {
             protocol.readBinary(sfi)
+        } catch (e: ISOFileNotFoundException) {
+            // Stop processing, the file doesn't exist.
+            return null
+        } catch (e: ISOSecurityStatusNotSatisfied) {
+            // Stop processing, the file is locked.
+            return null
         } catch (e: Exception) {
-            null
+            null // but continue
         }
 
         val records : MutableMap<Int, ImmutableByteArray> = mutableMapOf()
```

Comment:
I think this part and its dependencies can be committed. Not sure how I feel about skipping SFI dumping though